src/Sylius/Bundle/ApiBundle/Validator/Constraints/OrderProductInStockEligibilityValidator.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Validator\Constraints;
+
+use Sylius\Bundle\ApiBundle\Command\OrderTokenValueAwareInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Repository\OrderRepositoryInterface;
+use Sylius\Component\Inventory\Checker\AvailabilityCheckerInterface;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Webmozart\Assert\Assert;
+
+/** @experimental */
+final class OrderProductInStockEligibilityValidator extends ConstraintValidator
+{
+    /** @var OrderRepositoryInterface */
+    private $orderRepository;
+
+    /** @var AvailabilityCheckerInterface */
+    private $availabilityChecker;
+
+    public function __construct(
+        OrderRepositoryInterface $orderRepository,
+        AvailabilityCheckerInterface $availabilityChecker
+    ) {
+        $this->orderRepository = $orderRepository;
+        $this->availabilityChecker = $availabilityChecker;
+    }
+
+    /**
+     * @throws \InvalidArgumentException
+     */
+    public function validate($value, Constraint $constraint)
```

Comment:
```suggestion
    public function validate($value, Constraint $constraint)
```

I don't remember we were using in any other of validators this comment.